src/derivation.ts

```diff
@@ -144,11 +154,17 @@ export function validatePathSegment(
   let startsWithBip39 = false;
   path.forEach((node, index) => {
     if (index === 0) {
-      startsWithBip39 = BIP_39_PATH_REGEX.test(node);
-      if (!startsWithBip39 && !BIP_32_PATH_REGEX.test(node)) {
+      startsWithBip39 =
+        node instanceof Uint8Array || BIP_39_PATH_REGEX.test(node);
+
+      if (
+        !(node instanceof Uint8Array) &&
+        !startsWithBip39 &&
```

Comment:
Wouldn't it be enough to check if `!startsWithBip39`?

Reply:
The `node instanceof Uint8Array` is necessary for TypeScript:

```
TS2345: Argument of type 'BIP39Node | BIP32Node' is not assignable to parameter of type 'string'.   Type 'Uint8Array' is not assignable to type 'string'.
```

Reply:
Okay, can we leave a comment about that?